src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/MetadataTests/DefaultJsonTypeInfoResolverTests.JsonTypeInfo.cs

```diff
@@ -0,0 +1,695 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Text.Json.Serialization.Metadata;
+using System.Text.Json.Tests;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public static partial class DefaultJsonTypeInfoResolverTests
+    {
+        [Theory]
+        [InlineData(typeof(object))]
+        [InlineData(typeof(int))]
+        [InlineData(typeof(string))]
+        [InlineData(typeof(SomeClass))]
+        [InlineData(typeof(StructWithFourArgs))]
+        [InlineData(typeof(Dictionary<string, int>))]
+        [InlineData(typeof(DictionaryWrapper))]
+        [InlineData(typeof(List<int>))]
+        [InlineData(typeof(ListWrapper))]
+        public static void TypeInfoPropertiesDefaults(Type type)
```

Comment:
Maybe add a second argument holding the expected `JsonTypeInfoKind` and compare against the created instance?